src/main/client_context.cpp

```diff
@@ -355,6 +368,18 @@ std::unique_ptr<QueryResult> ClientContext::executeWithParams(PreparedStatement*
         std::string errMsg = exception.what();
         return queryResultWithError(errMsg);
     }
+    // rebind
+    KU_ASSERT(preparedStatement->parsedStatement != nullptr);
+    auto& parameterMap = preparedStatement->parameterMap;
+    preparedStatement->requiredNewTx = transactionContext->hasActiveTransaction() ? false : true;
+    try {
+        prepareInternal(*preparedStatement, false, "", parameterMap);
+    } catch (std::exception& exception) {
+        preparedStatement->success = false;
+        preparedStatement->errMsg = exception.what();
+        this->transactionContext->rollback();
```

Comment:
The more I read this code, the more concerned I have with our existing design. PrepareInternal opens a transaction but not commit/rollback it. Meaning whoever calls prepareInternal need to take care of the dangling tx.